include/types/ProfileUpdater.h

```diff
@@ -0,0 +1,143 @@
+   * @param b_below profile below
+   * @param cat category delimited by the profile (below)
+   * @param cat_above above category delimited by the profile
+   * @param ct_prof concordance table of the profile
+   * @param altPerf_model alternativePerformance calculated with current model
+   *
+   * @return the map of potential new perf for the profile and their
+   * desirability
+   */
+  std::unordered_map<float, float>
+  computeBelowDesirability(MRSortModel &model, std::string critId, Perf &b,
+                           Perf &b_below, Category &cat, Category &cat_above,
+                           std::unordered_map<std::string, float> &ct_prof,
+                           AlternativesPerformance &altPerf_model);
+
+  /**
+   * chooseMaxDesirability chooses the move of the profiles that maximizes the
+   * desirability
+   *
+   * @param desirability map of desirability
+   * @param b profile to move
+   *
+   */
+  float chooseMaxDesirability(std::unordered_map<float, float> &desirability,
+                              Perf &b);
+
+  /**
+   * updateTables updates model tables with new profile value
+   * - concordance table
+   * - good assignment count
+   * - alternative assignement
+   * - profiles
+   *
+   * @param model
+   * @param critId criterion on which the profile moves
+   * @param b_old old profile perf
+   * @param b_new new profile perf
+   * @param ct concordance table to update
+   * @param good # of good assignment to update
+   * @param altPerf_model alternativePerformance calculated with current model
+   *
+   */
+  void updateTables(
+      MRSortModel &model, std::string critId, Perf &b_old, Perf &b_new,
+      std::unordered_map<std::string, std::unordered_map<std::string, float>>
+          &ct,
+      int &good, AlternativesPerformance &altPerf_model);
+
+  /**
+   * optimizeProfile Optimizes one profile using the profileUpdater methods.
+   *
+   * @param prof profile to optimize
+   * @param cat_below category delimited by the profile (below)
+   * @param cat_above category delimited by the profile (above)
+   * @param model current model
+   * @param ct concordance table
+   * @param altPerf_model altPerf_model
+   *
+   */
+  void optimizeProfile(
+      std::vector<Perf> &prof, Category &cat_below, Category &cat_above,
+      MRSortModel &model,
+      std::unordered_map<std::string, std::unordered_map<std::string, float>>
+          &ct,
+      AlternativesPerformance &altPerf_model);
+
+  /**
+   * optimize Optimizes all the profiles using the profileUpdater methods.
+   *
+   * @param model current model
+   * @param ct concordance table
+   * @param altPerf_model altPerf_model
+   *
+   */
+  float optimize(MRSortModel &model,
+                 std::unordered_map<std::string,
+                                    std::unordered_map<std::string, float>> &ct,
+                 AlternativesPerformance &altPerf_model);
+
+private:
+  float epsilon_;
+  AlternativesPerformance &altPerf_data;
+  int good_;
```

Comment:
Actually the good variable shouldn't be in profile updater as the goal if this is to measure the right assigments and therefore select the best models. It might be useful to have is declared in the pipeline? However we need to thing of how we are going to count the right assignments at the end of a model improvement.

Reply:
we can also have a utils function that computes the CA like in PYMCDA

Reply:
Yes, that'll be implemented in the HeuristicPipeline

Reply:
I'll remove the good attribute from the class then, and we'll have the computation of the accuracy elsewhere. 
We should keep in mind that it might be useful to have the accuracy as an attribute from the model, so we don't need to recalculate the good/bad assignments at the end but we can update it directly in the ProfileUpdater.
The computation of good and bad assignment is already done in the ProfileUpdater it might loose efficiency to compute it again instead of just updating the value directly.

Reply:
Okay, let's add the equivalent of a score attribute in the model class and update its value in your algorithm then !

Reply:
Done, but we will need to initialize the value in the pipeline before going through the profile updater, alright?

Reply:
What do you mean? Can't the value be initialize at zero at the creation of the model?

Reply:
yes but at before going through the profile updater the score will not be 0!